src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -1809,6 +1809,23 @@ void IMergeTreeDataPart::makeCloneOnDisk(const DiskPtr & disk, const String & di
     volume->getDisk()->removeFileIfExists(fs::path(path_to_clone) / DELETE_ON_DESTROY_MARKER_FILE_NAME);
 }
 
+void IMergeTreeDataPart::makeCloneToOtherDisk(const DiskPtr & disk, const String & directory_name,const String & relative_data_path) const
+{
+    if (disk->getName() == volume->getDisk()->getName())
+        throw Exception("Can not clone data part " + name + " to same disk " + volume->getDisk()->getName(), ErrorCodes::LOGICAL_ERROR);
+    String path_to_clone = fs::path(relative_data_path) / directory_name;
+    if (disk->exists(path_to_clone))
+    {
+        LOG_WARNING(storage.log, "Path {} already exists. Will remove it and clone again.", fullPath(disk, path_to_clone));
+        disk->removeRecursive(fs::path(path_to_clone) / "");
```

Comment:
Are we sure that we don't delete any significant data? Maybe let's throw an exception in this case and let user to clean files manually.

Reply:
directory_name is a temp part name, path_to_clone is the path of directory_name,  besides, we have already checked the exsitence of  formal part path  in cloneAndLoadDataPart function,  so I think we can remove this temp folder here.